features/collection/api/src/main/java/org/opennms/netmgt/collection/support/ObjectNameStorageStrategy.java

```diff
@@ -0,0 +1,138 @@
+    private static final String PARAM_CLEAN_OUTPUT = "clean-output";
+    private String m_indexFormat;
+    private Boolean m_cleanOutput = false;
+
+    private static final JexlEngine JEXL_ENGINE;
+    private static final UnifiedJEXL EL;
+
+    static {
+        final int cacheSize = Integer.getInteger("org.opennms.netmgt.collection.support.ObjectNameStorageStrategy.cacheSize", DEFAULT_JEXLENGINE_CACHESIZE);
+        JEXL_ENGINE = new JexlEngine();
+        JEXL_ENGINE.setCache(cacheSize);
+        JEXL_ENGINE.setLenient(false);
+        JEXL_ENGINE.setStrict(true);
+
+        EL = new UnifiedJEXL(JEXL_ENGINE);
+    }
+
+    public ObjectNameStorageStrategy() {
+        super();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getResourceNameFromIndex(CollectionResource resource) {
+        String resourceName = null;
+        try {
+            ObjectName oname = new ObjectName(resource.getInstance());
+            UnifiedJEXL.Expression expr = EL.parse( m_indexFormat );
+
+            JexlContext context = new MapContext();
+            context.set("ObjectName", oname);
+            context.set("domain", oname.getDomain() == null ? "" : oname.getDomain());
+            oname.getKeyPropertyList().entrySet().forEach((entry) -> {
+                final String value = entry.getValue();
+                if (value.startsWith(QUOTE) && value.endsWith(QUOTE)) {
+                    context.set(entry.getKey(), ObjectName.unquote(entry.getValue()));
+                } else {
+                    context.set(entry.getKey(), entry.getValue());
+                }
+            });
+            resourceName = (String) expr.evaluate(new ReadonlyContext(context));
+        } catch (JexlException e) {
+            LOG.error("getResourceNameFromIndex(): error evaluating index-format [{}] as a Jexl Expression", m_indexFormat, e);
+        } catch (javax.management.MalformedObjectNameException e) {
+            LOG.debug("getResourceNameFromIndex(): malformed object name: {}", resource.getInstance(), e);
+        } finally {
+            if (resourceName == null) {
+                resourceName = resource.getInstance();
+                resourceName = resourceName.replaceAll("\\s+", "_").replaceAll(":", "_").replaceAll("\\\\", "_").replaceAll("[\\[\\]]", "_").replaceAll("[|/]", "_").replaceAll("=", "").replaceAll("[_]+$", "").replaceAll("___", "_");
+            }
+        }
+        if (m_cleanOutput) {
+            resourceName = resourceName.replaceAll("\\s+", "_").replaceAll(":", "_").replaceAll("\\\\", "_").replaceAll("[\\[\\]]", "_").replaceAll("[|/]", "_").replaceAll("=", "").replaceAll("[_]+$", "").replaceAll("___", "_");
+        }
+
+        LOG.debug("getResourceNameFromIndex(): {}", resourceName);
+        return resourceName;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setParameters(List<Parameter> parameterCollection) throws IllegalArgumentException {
+        if (parameterCollection == null) {
```

Comment:
I was just following what was done in SiblingColumnStorageStrategy. Otherwise you need to wrap` Objects.requireNonNull(parameterCollection) ` with a try statement to catch the NullPointerException it throws when null to re-throw it as a IllegalArgumentException.